File: estoque/model/produto.go
package model

import "gorm.io/gorm"

type Produto struct {
	gorm.Model
	Codigo    string  `json:"codigo" gorm:"uniqueIndex;not null"`
	Descricao string  `json:"descricao" gorm:"not null"`
	Saldo     float64 `json:"saldo" gorm:"not null;default:0"`
}

File: frontend/src/app/services/produto.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Produto {
  id: number;
  codigo: string;
  descricao: string;
  saldo: number;
}

@Injectable({
  providedIn: 'root'
})
export class ProdutoService {
  private apiUrl = 'http://localhost:8080';

  constructor(private http: HttpClient) {}

  listar(): Observable<Produto[]> {
    return this.http.get<Produto[]>(`${this.apiUrl}/produtos`);
  }

  criar(produto: Omit<Produto, 'id'>): Observable<Produto> {
    return this.http.post<Produto>(`${this.apiUrl}/produtos`, produto);
  }
}